search-sentiments-map/routes/search.js

```diff
@@ -26,22 +26,141 @@ const json = require('./../public/country-code.json');
     res.send(customSearchTopicJsonArray);
   });
 });
 
+/** 
+ * Renders a JSON array of the top search results for requested countries with
+ * API data from within the last 12 hours for the specified topic.
+ */
+router.get('/:topic/:countries', (req, res) => {
+  let topic = req.params.topic;
+  let countries = JSON.parse(req.params.countries);
+  retrieveUserSearchResultFromDatastore(topic, countries).then(userSearchTopicJsonArray => {
+    res.setHeader('Content-Type', 'application/json');
+    res.send(userSearchTopicJsonArray);
+  });
+});
+
+/** 
+ * Retrieves search result data for specified countries for a given topic.
+ *   - Obtains and formats search results from Custom Search API if the data
+ *         is not currently stored.
+ *   - Saves any new search results to the datastore.
+ *   - Retrieves any relevant existing search results from datastore.
+ *   - Returns all requested search results on the topic from the countries.
+ * @param {string} topic Search topic to get data for.
+ * @param {Array} countries Countries to get search topic data for.
+ * @return {Object} JSON-formatted array of search results from requested
+ *     countries.
+ */
+async function retrieveUserSearchResultFromDatastore(topic, countries) {
+  // Request latest entity with a topic matching the given topic.
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp', {
+    descending: true,
+  }).filter('lowercaseTopic', topic.toLowerCase()).limit(1);
+  const [customSearchTopic] = await datastore.runQuery(query);
+
+  let countriesDataToReturn = [];
+  let timestamp;
+  if (customSearchTopic.length !== 0 &&
+      Date.now() - customSearchTopic[0].timestamp <
+      CURRENT_SEARCH_RESULT_THRESHOLD_24_HOURS_MS) {
+        timestamp = customSearchTopic[0].timestamp;
+        let countriesToAddDataFor = [];
+        let countriesData = customSearchTopic[0].countries;
+
+        // Determine whether a country has existing data or data needs to be
+        // retrieved from Custom Search API for this country on this topic.
+        countries.forEach(country => {
+          let countryData = countriesData
+              .filter(countries => countries.country === country);
+          if (countryData.length === 0) {
+            countriesToAddDataFor.push(country);
+          } else {
+            countriesDataToReturn.push(countryData[0]);
+          }
+        });
+
+        // Obtain custom search data for countries without current data.     
+        // Add new custom search data to existing entity and to the data to
+        // send back to front end.
+        if (countriesToAddDataFor.length !== 0) {
+          let newCountriesData = await getSearchResultsForArrayOfCountries(
+            countriesToAddDataFor, topic);
+          await addNewCountryData(newCountriesData, customSearchTopic[0]);
+          countriesDataToReturn = countriesDataToReturn.concat(newCountriesData);
+        }
```

Comment:
Indentation seems too much to the right.

Reply:
I wasn't sure because the if conditional is indented 4 spaces. Should I move this to instead be only 2 spaces indented from the original if?